sympy/polys/rings.py

```diff
@@ -1953,20 +1954,44 @@ def imul_num(p, c):
             p[exp] *= c
         return p
 
-    def content(f):
-        """Returns GCD of polynomial's coefficients. """
-        domain = f.ring.domain
-        cont = domain.zero
+    def content(self, domain):
+        """Returns GCD of polynomial's coefficients.
+
+        Examples
+        ========
+
+        >>> from sympy.polys import ring, ZZ
+        >>> R, x = ring("x", ZZ)
+
+        >>> f = 3*x**2 + 12*x + 6
+        >>> f.content(ZZ)
+        3
+
+        >>> f = [3, 12, 6]
+        >>> f.content(ZZ)
+        3
+
+        """
+        f = self
+        if not isinstance(f, list):
+            f = f.listcoeffs()
+
+        coeffs = [int(x) for x in f]
         gcd = domain.gcd
+        cont = coeffs[0]
 
-        for coeff in f.itercoeffs():
+        for coeff in coeffs[1:]:
             cont = gcd(cont, coeff)
 
+            if cont == domain.one:
+                break
+
         return cont
```

Comment:
There is an error that `list has no attribute content`
Where should I add this method to the list type?